From cf6cf2060c5a7a7ddc0396a0c20c234fc56c79b6 Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei@gherzan.ro>
Date: Tue, 28 May 2019 18:02:24 +0100
Subject: [PATCH] Don't abort gst_omx_video_dec_set_format() if there's a
 timeout releasing the buffers taken by the egl_render out port

From 0d2ad639e6158c8023c157e206ef3ff7abdc089c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?=
<eocanha@igalia.com>
Date: Fri, 4 Dec 2015 18:39:59 +0100
Subject: [PATCH] Don't abort gst_omx_video_dec_set_format() if there's a
timeout releasing the buffers taken by the egl_render out port

Upstream-status: Pending
Signed-off-by: Andrei Gherzan <andrei@gherzan.ro>
---
 omx/gstomxvideodec.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index ba5304f..8bd5d3d 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -2229,7 +2229,9 @@ gst_omx_video_dec_disable (GstOMXVideoDec * self)
       return FALSE;
     if (gst_omx_port_wait_buffers_released (out_port,
             1 * GST_SECOND) != OMX_ErrorNone)
+#if !(defined (USE_OMX_TARGET_RPI) && defined (HAVE_GST_GL))
       return FALSE;
+#endif
     if (!gst_omx_video_dec_deallocate_output_buffers (self))
       return FALSE;
     if (gst_omx_port_wait_enabled (out_port, 1 * GST_SECOND) != OMX_ErrorNone)
-- 
2.17.1

